src/core/io/processinterfacefb.cpp

```diff
@@ -0,0 +1,253 @@
+    CEventSourceFB(paContainer, paInterfaceSpec, paInstanceNameId),
+    IOObserver(),
+    var_QI(0_BOOL),
+    var_PARAMS(""_STRING),
+    var_QO(0_BOOL),
+    var_STATUS(""_STRING),
+    var_conn_QO(var_QO),
+    var_conn_STATUS(var_STATUS),
+    conn_INITO(this, 0),
+    conn_CNF(this, 1),
+    conn_QI(nullptr),
+    conn_PARAMS(nullptr),
+    conn_QO(this, 0, &var_conn_QO),
+    conn_STATUS(this, 1, &var_conn_STATUS) {
+  mIsListening = false;
+  mIsReady = false;
+}
+
+void CProcessInterfaceFB::setInitialValues() {
+  var_QI = 0_BOOL;
+  var_PARAMS = ""_STRING;
+  var_QO = 0_BOOL;
+  var_STATUS = ""_STRING;
+}
+
+void CProcessInterfaceFB::readInputData(const TEventID paEIID) {
+  switch(paEIID) {
+    case scmEventINITID: {
+      readData(0, var_QI, conn_QI);
+      readData(1, var_PARAMS, conn_PARAMS);
+      break;
+    }
+    case scmEventREQID: {
+      readData(0, var_QI, conn_QI);
+      break;
+    }
+    default:
+      break;
+  }
+}
+
+void CProcessInterfaceFB::writeOutputData(const TEventID paEIID) {
+  switch(paEIID) {
+    case scmEventINITOID: {
+      writeData(0, var_QO, conn_QO);
+      writeData(1, var_STATUS, conn_STATUS);
+      break;
+    }
+    case scmEventCNFID: {
+      writeData(0, var_QO, conn_QO);
+      writeData(1, var_STATUS, conn_STATUS);
+      break;
+    }
+    default:
+      break;
+  }
+}
+
+CProcessInterfaceFB::~CProcessInterfaceFB() {
+  deinitialise();
+}
+
+bool CProcessInterfaceFB::initialise(bool paIsInput, CEventChainExecutionThread *const paECET) {
+  mDirection = paIsInput ? IOMapper::In : IOMapper::Out;
+  if(paIsInput && (getFBInterfaceSpec().mNumDOs < 3)) {
+    mType = CIEC_ANY::e_Max; //we assume that any FB which has no "IN" Output must be a EVENT-Only FB.
+  }
+  else {
+    //as it has a index 2 here, we safely can do this
+    mType = (paIsInput ? getDO(2) : getDI(2))->getDataTypeID();
+  }
+
+  mIsReady = false;
+  var_STATUS = scmWaitingForHandle;
+
+  // Reset before initialization
+  deinitialise();
+
+  // Register interface
+  if(!(mIsListening = IOMapper::getInstance().registerObserver(getId(), this))) {
+    var_STATUS = scmFailedToRegister;
+    return false;
+  }
+
+  if(mDirection == IOMapper::In) {
+    setEventChainExecutor(paECET);
+  }
+
+  return mIsReady;
+}
+
+// If PARAMS is not empty use the string specified here "as is" for the IO mapping to the configuration.
+// Otherwise use the full qualified instance name of our FB.
+std::string CProcessInterfaceFB::getId() {
+  if (!var_PARAMS.empty()) {
+    return static_cast<std::string>(var_PARAMS);
+  }
+  else {
+    return getFullQualifiedApplicationInstanceName('.');
+  }
+}
+
+bool CProcessInterfaceFB::deinitialise() {
+  // Deregister interface
+  if(mIsListening) {
+    IOMapper::getInstance().deregisterObserver(this);
+    mIsListening = false; // we are deregistered ! 
+  }
+
+  return !mIsReady;
+}
+
+CIEC_BOOL CProcessInterfaceFB::read(CIEC_ANY &paData) {
```

Comment:
I think paData could be const

Reply:
no because we read from the process interface to store it in paData.

Reply:
Oh, my bad. Thats a read from to paData? How about the write method below then? That's then setting the writing the value from paData to the IO?